### PR TITLE
chore: add forge, linter and prettier

### PR DESCRIPTION
**Pull request by vojtechsimetka**
TODO:
- [x] add typechain
- [x] add README with local test setup guide
- [x] add prettier

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Use Node.js ${{ env.NODE_VERSION }}
+        uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
```
Comment (filoozom) [inline .github/workflows/test.yml:33]: Does it have to be nightlies? Feels like that's unnecessarily unstable.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+      - name: Start blockchain
+        run: |
+          anvil &
```
Comment (filoozom) [inline .github/workflows/test.yml:37]: Perfect, that's what I wanted to use as well!

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+interface CreateMarketplace {
+    marketplace: Marketplace
+    providerRepToken: MintableERC20
+    seekerRepToken: MintableERC20
+}
+
+export async function createMarketplace(marketplaceFactory: MarketplaceFactory, marketplaceTokenAddress: string, marketplaceName: string, marketplaceFee: BigNumberish, metadata: string): Promise<CreateMarketplace> {
+        let promise = new Promise<EventMarketplaceCreatedData>((resolve) => {
```
Comment (filoozom) [inline src/index.ts]: Could be a `const` I guess?

File: test/index.spec.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { WebSocketProvider } from '@ethersproject/providers'
+import { BigNumberish, Wallet } from 'ethers'
+import {getMarketplaceFactory, getERC20, createMarketplace } from '../src/index'
```
Comment (filoozom) [inline test/index.spec.ts]: Some issue with `prettier`, is it set up (correctly)?

**Review by filoozom — COMMENTED** [review]
I love it! `forge` + `anvil` + `typechain` feels like the future for sure.

**Review by filoozom — APPROVED** [review]

File: test/index.spec.ts (hunk 4 above)
Comment (filoozom) [reply test/index.spec.ts]: (Got my answer in the message, not setup yet, no problemo.)

File: .github/workflows/test.yml (hunk 1 above)
Comment (vojtechsimetka) [reply .github/workflows/test.yml:33]: Sadly there is no 1.0.0 version only nightly for the time being :( https://github.com/foundry-rs/foundry-toolchain

**Pull request merged**